### PR TITLE
TDD Project

### PR DESCRIPTION
**Pull request by raminka13**
Ruby

- Created a class called `Solver`.
- Created a method called `factorial` that takes one argument, an integer `N`, and returns the [factorial](https://en.wikipedia.org/wiki/Factorial) for that number. The factorial is the multiplication of all integers from 1 to `N` and has the special case that the factorial of 0 is 1. This method only accepts 0 and positive integers, so if a negative integer is given it should raise an exception.
- Created a method called `reverse` that takes one argument, a string `word`, and returns `word` reversed (e.g. if `word` is `"hello"` it returns `"olleh"`).
- Created a method called `fizzbuzz` that takes one argument, an integer `N`, and returns a string. The returned string is constructed following these rules:
  - When `N` is divisible by 3, return `"fizz"`.
  - When `N` is divisible by 5, return `"buzz"`.
  - When `N` is divisible by 3 and 5, return `"fizzbuzz"`.
  - In any other case, return `N` as a string (e.g. say `N` is `7` then return `"7"`).
- Remember to start with tests for all of the methods described above. Your commit history should prove that you have used TDD.

### CODE REVIEW

File: spec/solver_spec.rb [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+    it 'should return 362,880 for the factorial of 9' do
+      result = @solver.factorial(9)
+      expect(result).to eq 362_880
+    end
+    it 'should return 3,628,800 for the factorial of 10' do
+      result = @solver.factorial(10)
+      expect(result).to eq 3_628_800
+    end
+  end
+
```
Comment (julie-ify) [inline spec/solver_spec.rb]: - [x] For the Factorial testing, kindly include a scenario where the number is negative, and it should expect to raise an exception.

File: spec/solver_spec.rb [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+    it 'should return Buzz if divisible by 5' do
+      result = @solver.fizzbuzz(5)
+      expect(result).to eq 'Buzz'
+    end
+
+    it 'should return FizzBuzz if divisible by 3 and 5 at the same time' do
+      result = @solver.fizzbuzz(15)
+      expect(result).to eq 'FizzBuzz'
+    end
+  end
```
Comment (julie-ify) [inline spec/solver_spec.rb]: - [x] For the FizzBuzz testing you need to include a scenario for when the value is neither divisible by 3 nor 5. For example, if the number 7 is supplied it should return the string value: '7'.

**Review by julie-ify — CHANGES_REQUESTED** [review]
## Hi @raminka13 and @for-ashraf,

Good job so far!

## Highlights!
- Good feature branch name ✔️ 
- No linter errors ✔️ 
- Descriptive pull request ✔️ 
- Descriptive commit messages ✔️ 
- Good code organization and working app ✔️
- Descriptive README ✔️ 

Hello ✋, you have done well reaching this point of the milestone 🤝 . But there are some issues that you still need to work on to go to the next project but you are almost there!

# Status: Required Changes ♻️⛔️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread and tag me `@julie-ify` if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: spec/solver_spec.rb (hunk 1 above)
Comment (raminka13) [reply spec/solver_spec.rb]: Thanks for the correction @julie-ify

Comment (rloterh) [reply spec/solver_spec.rb]: well resolved! ✔️

File: spec/solver_spec.rb (hunk 2 above)
Comment (rloterh) [reply spec/solver_spec.rb]: well resolved! ✔️

Comment (raminka13) [reply spec/solver_spec.rb]: Hey @rloterh and @julie-ify,
I just pushed the changes, they resolved the missing scenarios.
Thanks.

**Review by rloterh — APPROVED** [review]
## Status: Approved :heavy_check_mark:

Hi Muhammad & Raminka,  

Your project is complete! There is nothing else to say other than... it's time to merge it 🍾. Congratulations!  🎉💯🌟 

## Highlights 👍
- Properly implemented unit tests with Rspec
- Good ReadMe

<br>
Keep soaring higher! 💻 🍷 🚀 
<br><br>
Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**